README.md

```diff
@@ -49,6 +49,30 @@ The runtime library provides the following:
 * Unit test, coverage, and trace support code.
 * Low-level helpers for compiler-inserted calls.
 
+Source structure
+------
+
+In the `src` directory, there are `core` and `rt` packages.
+
+The `rt` package contains implementations for compiler-inserted calls.
+You may not `import rt.xxx` outside of the `rt` package itself, since its source is not accessible once druntime is compiled;
+it isn't included in the `druntime/import` that ships with compiler releases.
+The compiler assumes these functions have a specific `extern(C)` signature, and expects them to be linked into the final binary.
+See for example: [https://github.com/dlang/dmd/blob/master/src/dmd/backend/drtlsym.d](dmd/backend/drtlsym.d)
+
+The `core` package is shipped with the compiler and has a stable API.
+It contains implementations for D features, but also some D conversions of header files from external libraries:
+- `core/stdc` - C standard library
+- `core/stdcpp` - C++ standard library
+- `core/sys` - operating system API
+
+An exception to the public API is `core.internal`, which contains compiler hooks that are templates.
```

Comment:
`core.internal` also contains the gc implementation and utilities for backtraces etc.

The GC isn't technically part of the public API, but if you want to inherit from the real GC you have to get ConvervativeGC from inside `core.internal`